File: Search/maven/src/test/java/Search/ProcessorTest.java
package Search;

import static org.junit.Assert.assertArrayEquals;
import static org.junit.Assert.assertEquals;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;
import java.util.ArrayList;
import org.junit.Before;
import org.junit.Test;


public class ProcessorTest {

    private String[] docs ;
    private Processor processor ;


   
    
    @Before
    public void init() {

         docs = new String[3];
         docs[0] = "The sky , at sunset , looked like a carnivorous flower ." ;
         docs[1] = "Let the sun stay beautiful in the blue sky ." ;
         docs[2] = "Trees are poems that the earth writes upon the sky ." ;

         FileReader fileReader = mock(FileReader.class);
         when(fileReader.hasNext()).thenReturn(true,true,true,false);
         when(fileReader.next()).thenReturn(docs[0],docs[1],docs[2]);

         processor  =  new Processor(fileReader);

    }
    
    @Test
    public void fillTheMapTest() {
        
        int numberOfDocs = processor.fillTheMap();
        assertEquals(3, numberOfDocs);
   

    }

    @Test
    public void processTest1(){

        fillTheMapTest();
        ArrayList<Integer> actualValue = processor.process("sky -blue -flower", 3);
        Integer[] expectedValue = {2};

        assertArrayEquals(expectedValue, actualValue.toArray());

    }

    @Test
    public void processTest2(){

        fillTheMapTest();
        ArrayList<Integer> actualValue = processor.process("+flower +trees", 3);
        Integer[] expectedValue = {0,2};

        assertArrayEquals(expectedValue, actualValue.toArray());

    }
    
    @Test
    public void processTest3(){

        fillTheMapTest();
        ArrayList<Integer> actualValue = processor.process("-poems -earth -sun", 3);
        Integer[] expectedValue = {1,2};

        assertArrayEquals(expectedValue, actualValue.toArray());

    }



    }
   

File: Phase05/Search/Search.Test/UtilsTest/ProcessorTest.cs
using System;
using System.Collections.Generic;
using Moq;
using Search.Model;
using Search.Utils;
using Xunit;

namespace Search.Test.UtilsTest
{
    public class ProcessorTest
    {

        private string[] docs;
        private Processor Processor;


        public void Init()
        {

            docs = new String[3];
            docs[0] = "The sky , at sunset , looked like a carnivorous flower .";
            docs[1] = "Let the sun stay beautiful in the blue sky .";
            docs[2] = "Trees are poems that the earth writes upon the sky .";

            var enumerator = docs.GetEnumerator();

            var fileReader = new Mock<IReader>();

            fileReader.Setup(x => x.MoveNext()).Returns(enumerator.MoveNext);
            fileReader.Setup(x => x.Current()).Returns(() => (string)enumerator.Current);

            Processor = new Processor(fileReader.Object);

        }

        [Fact]
        public void FillTheMapTest()
        {

            Init();
            int numberOfDocs = Processor.FillTheMap();
            Assert.Equal(3, numberOfDocs);


        }

        [Fact]
        public void processTest1()
        {

            FillTheMapTest();
            var actualValue = Processor.Process("sky -blue -flower", 3);
            var expectedValue = new HashSet<int> { 2 };

            Assert.Equal(expectedValue, actualValue);

        }

        [Fact]
        public void processTest2()
        {

            FillTheMapTest();
            var actualValue = Processor.Process("+flower +trees", 3);
            var expectedValue = new HashSet<int> { 0, 2 };

            Assert.Equal(expectedValue, actualValue);


        }

        [Fact]
        public void processTest3()
        {
            FillTheMapTest();
            var actualValue = Processor.Process("-poems -earth -sun", 3);
            var expectedValue = new HashSet<int> { 0 };

            Assert.Equal(expectedValue, actualValue);

        }

    }

}
